superset-frontend/src/dashboard/util/getRelatedCharts.ts

```diff
@@ -80,12 +80,25 @@ export function getRelatedCharts(
   datasources: DatasourcesState,
 ) {
   return Object.entries(filters).reduce((acc, [filterKey, filter]) => {
+    const chartsInScope = filter.chartsInScope
```

Comment:
`filter.chartsInScope?.filter(
           (chartId: number) => !filter.scope.excluded.includes(chartId),
         ) ?? filter.scope`

Reply:
Thanks. I think from looking at the unit tests there is an issue with this logic too. I am fixing it up.